packages/netlify-cms-core/src/reducers/editorialWorkflow.ts

```diff
@@ -121,14 +132,23 @@ const unpublishedEntries = (state = Map(), action: EditorialWorkflowAction) => {
       );
 
     case UNPUBLISHED_ENTRY_PUBLISH_SUCCESS:
+      return state.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
+
     case UNPUBLISHED_ENTRY_PUBLISH_FAILURE:
-      return state.withMutations(map => {
-        map.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
-      });
+      return state.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
+
+    case UNPUBLISHED_ENTRY_DELETE_REQUEST:
+      return state.setIn(
+        ['entities', `${action.payload!.collection}.${action.payload!.slug}`, 'isDeleting'],
+        true,
+      );
 
     case UNPUBLISHED_ENTRY_DELETE_SUCCESS:
       return state.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
 
+    case UNPUBLISHED_ENTRY_DELETE_FAILURE:
```

Comment:
If the delete failed I'm not sure we should delete the entry from the local state.

Reply:
So I can change that part like this:
```
 return state.setIn(
        ['entities', `${action.payload!.collection}.${action.payload!.slug}`, 'isDeleting'],
        false,
      );
```

WDYT?

Reply:
Yes, that would work. I would actually keep the old code (not handling that action all-together) since I don't think we read `isDeleting` at the moment

Reply:
So it is gonna be like that:

```
return state.withMutations(map => {
        map.setIn(
          ['entities', `${action.payload!.collection}.${action.payload!.slug}`, 'isDeleting'],
          false,
        );
        map.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);

      });
```

ok?